### PR TITLE
[ELYEE-5] Merge changes from wildfly-elytron and split out the classes from the org.wildfly.security.auth.client.spi package

### PR DESCRIPTION
**Pull request by fjuma**
https://issues.redhat.com/browse/ELYEE-5
https://issues.redhat.com/browse/ELYEE-6
https://issues.redhat.com/browse/ELYEE-7

This PR does the following:
* Merges the latest changes from the wildfly-elytron 1.x branch (to pull in the updates to `JaspiAuthenticationContext`)
* Merges the 2.x branch and splits out the classes from the `org.wildfly.security.auth.client.spi`
* Upgrades to the 2.x version of Elytron

(This PR depends on the upcoming 2.0.0.Beta1 release of Elytron)

### CODE REVIEW

**Review by skyllarr — APPROVED** [review]

File: ELY_Messages.txt [hunk 1]
```diff
@@ -82,6 +82,7 @@
 01041 - 01153    wildfly-elytron-realm-ldap
 04025            wildfly-elytron-realm-ldap
 01104 - 01180    wildfly-elytron-realm-token
+01181 - 01185    wildfly-elytron-realm-token
```
Comment (darranl) [inline ELY_Messages.txt:85]: It doesn't need handling in this PR but this reminds me we probably need to check the individual projects now have their independent version ranges reserved so we don't need to worry about too much cross project coordination.

**Review by darranl — APPROVED** [review]

**Comment by darranl** [comment]
@fjuma this may need checking if the switch to ubunto-latest is the cause for the failure?

File: ELY_Messages.txt (hunk 1 above)
Comment (fjuma) [reply ELY_Messages.txt:85]: Yes, good point.

**Comment by fjuma** [comment]
There was an issue with ubuntu-latest and the jmockit version. Upgrading the jmockit version fixes that failure. However, it turns out that nothing in this repo actually makes use of jmockit so to clean things up, I've removed that dependency.

**Pull request merged**